pkg/telemetry/telemetry.go

```diff
@@ -293,6 +294,28 @@ func (t *engineImpl) housekeeping() {
 		return
 	}
 	t.writeToTimescaleMetadataTable(stats)
+	t.syncSQLStats()
+}
+
+func (t *engineImpl) syncSQLStats() {
+	pending := make(Metadata)
+	var err error
+	for _, stat := range sqlTelemetry {
+		output := new(string)
+		err = t.conn.QueryRow(context.Background(), stat.query).Scan(&output)
+		if err != nil || output == nil {
+			// Do not stop. Let's try to collect those stats that do not error out,
+			// so we have something to report.
+			//
+			// A nil output signifies an empty response.
+			continue
+		}
+		pending[stat.name] = *output
+	}
+	if err != nil {
+		log.Debug("msg", "some SQL stats did not complete successfully", "last err", err.Error())
+	}
+	t.writeToTimescaleMetadataTable(pending)
```

Comment:
This feels like a very "chatty" way to do this. Why not write a database function to collect all these metrics and insert them into the metadata table? Or at least write a database function that returns a set of text and gets all the metrics in one database round-trip?